### PR TITLE
proto2spice: Vlsir proto messages to Spice netlists through VlsirTools netlister

### PR DESCRIPTION
**Pull request by growly**
depends on https://github.com/Vlsir/Vlsir/pull/79, whose changes are subsumed here

### CODE REVIEW

**Comment by dan-fritchman** [comment]
Did add all these JavaScript & C++ bindings get added back on purpose? (I expect not.)

**Comment by growly** [comment]
js/cpp bindings were added by accident in #79, and have been removed there and here

File: protos/utils.proto [hunk 1]
```diff
@@ -36,11 +36,11 @@ enum SIPrefix {
...
 message Prefixed {
-  SIPrefix prefix = 1;  // The metric `SIPrefix`
-  oneof number {        // The numeric suffix, either as int, float, or string
-    int64 integer = 2;  // Int
-    double double = 3;  // Float
-    string string = 4;  // String
+  SIPrefix prefix = 1;          // The metric `SIPrefix`
+  oneof number {                // The numeric suffix, either as int, float, or string
+    int64 int64_value = 2;
+    double double_value = 3;
```
Comment (dan-fritchman) [inline protos/utils.proto:42]: Long discussed: 
Is this the time to dump the floating-point variant here? 
It's kinda antithetical to the point of the thing.

File: VlsirTools/vlsirtools/netlist/spice.py [hunk 2]
```diff
@@ -143,6 +143,9 @@ def write_instance_name(
     def write_instance(self, pinst: vckt.Instance) -> None:
         """Create and return a netlist-string for Instance `pinst`"""
 
+        if not pinst.HasField("module"):
```
Comment (dan-fritchman) [inline VlsirTools/vlsirtools/netlist/spice.py:146]: What's this do? 
Is that like a semi-recently deprecated field name?

File: protos/utils.proto (hunk 1 above)
Comment (growly) [reply protos/utils.proto:42]: We can discuss that at the next meeting. I will remove it and send the PR if we decide to. Prefer not to overload this change.

File: VlsirTools/vlsirtools/netlist/spice.py (hunk 2 above)
Comment (growly) [reply VlsirTools/vlsirtools/netlist/spice.py:146]: (From experience) if your incoming vckt.Package is garbage and has Instances that don't refer to Modules (i.e. the Module field is empty), `write_instance` fails during the call to `resolve_reference`, and it isn't really clear why. This just provides a helpful error for users feeding in the garbage to know what to do about it.

**Comment by dan-fritchman** [comment]
Couple small (and/or not necessarily on-task) comments but LGTM. 

Also helpful: stick the script in something like `VlsirTools/scripts/`, and then folks who `pip install vlsirtools` get it installed right into their PATH: 
https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#scripts

File: VlsirTools/vlsirtools/netlist/spice.py (hunk 2 above)
Comment (dan-fritchman) [reply VlsirTools/vlsirtools/netlist/spice.py:146]: I see, it's just like the null-check before passing to something that expects an (actual) `Ref`. Makes sense!

**Comment by growly** [comment]
moved to scripts/, will wait til #79 is merged to merge this

**Comment by dan-fritchman** [comment]
Sure. Or just skip 79 (this includes all its changes right)?

**Comment by dan-fritchman** [comment]
Created #81 for the script-publishing noted above.  
I believe this should just require a setup.py option or two.

**Pull request merged**